internal/dag/builder_test.go

```diff
@@ -28,8 +30,375 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
 )
 
+func TestDAGInsertServiceAPIs(t *testing.T) {
+
+	svc1 := &v1.Service{
```

Comment:
I'd recommend calling these kuardSvc and blogSvc respectively, that saves one lookup when you're writing more tests.